### PR TITLE
Add microsoft teams notification provider

### PR DESCRIPTION
**Pull request by WillianRod**
## What I've done?
- Added Microsoft Teams notification provider

## Screen captures
image

#### Test notification
image

#### Down notification
image

#### Up notification
image

### CODE REVIEW

**Comment by gaby** [comment]
@WillianRod Microsoft Teams is already supported via `apprise` (already preinstalled in the Docker Image).

More information here: https://github.com/caronc/apprise/wiki/Notify_msteams

**Comment by WillianRod** [comment]
> 
> 
> @WillianRod Microsoft Teams is already supported via `apprise` (already preinstalled in the Docker Image).
> 
> More information here: https://github.com/caronc/apprise/wiki/Notify_msteams

@gabyI know that, but in my use-case (running Uptime Kuma with Caprover) it would be more pratical if teams integration was built in.

Where I work we have a total of 12 teams, my intention is to bring Kuma to be our main monitoring tool. We use Microsoft Teams to chat and each team have its own Microsoft Teams channel.

My intention is to have all teams webhook url seted up and when adding a new monitor url, I just need to select the Microsoft Teams notification for the team responsible for that aplication.

image

| Application | Team |
| --- | --- |
| Application 01 | Team A |
| Application 02 | Team A |
| Application 03 | Team B |
| Application 04 | Team C |
| Application 05 | Team C |

With Microsoft Teams integration built in I just need to create a webhook url for every new team and add it to Kuma notifications.

**Comment by MichelBaie** [comment]
I totally agree and would like to do same

**Comment by gaby** [comment]
> > @WillianRod Microsoft Teams is already supported via `apprise` (already preinstalled in the Docker Image).
> > More information here: [caronc/apprise/wiki/Notify_msteams](https://github.com/caronc/apprise/wiki/Notify_msteams)
> 
> @gaby I know that, but in my use-case (running Uptime Kuma with Caprover) it would be more pratical if teams integration was built in.
> 
> Where I work we have a total of 12 teams, my intention is to bring Kuma to be our main monitoring tool. We use Microsoft Teams to chat and each team have its own Microsoft Teams channel.
> 
> My intention is to have all teams webhook url seted up and when adding a new monitor url, I just need to select the Microsoft Teams notification for the team responsible for that aplication.
> 
> image
> 
> Application	Team
> Application 01	Team A
> Application 02	Team A
> Application 03	Team B
> Application 04	Team C
> Application 05	Team C
> With Microsoft Teams integration built in I just need to create a webhook url for every new team and add it to Kuma notifications.

I see what you are trying to do, I just don't see how is that different than creating multiple notifications with apprise? Apprise is already built-in as you said, why should uptime-kuma create its own implementation of a notification already supported.

**Comment by MichelBaie** [comment]
> > > @WillianRod Microsoft Teams is already supported via `apprise` (already preinstalled in the Docker Image).
> > > More information here: [caronc/apprise/wiki/Notify_msteams](https://github.com/caronc/apprise/wiki/Notify_msteams)
> > 
> > 
> > @gaby I know that, but in my use-case (running Uptime Kuma with Caprover) it would be more pratical if teams integration was built in.
> > Where I work we have a total of 12 teams, my intention is to bring Kuma to be our main monitoring tool. We use Microsoft Teams to chat and each team have its own Microsoft Teams channel.
> > My intention is to have all teams webhook url seted up and when adding a new monitor url, I just need to select the Microsoft Teams notification for the team responsible for that aplication.
> > image
> > Application	Team
> > Application 01	Team A
> > Application 02	Team A
> > Application 03	Team B
> > Application 04	Team C
> > Application 05	Team C
> > With Microsoft Teams integration built in I just need to create a webhook url for every new team and add it to Kuma notifications.
> 
> I see what you are trying to do, I just don't see how is that different than creating multiple notifications with apprise? Apprise is already built-in as you said, why should uptime-kuma create its own implementation of a notification already supported.

I just want Microsoft Teams, not other services, why not integrate it directly into uptime-kuma ?
Teams is for professional use, and i think lot peoples are using it

**Comment by gaby** [comment]
> > > > @WillianRod Microsoft Teams is already supported via `apprise` (already preinstalled in the Docker Image).
> > > > More information here: [caronc/apprise/wiki/Notify_msteams](https://github.com/caronc/apprise/wiki/Notify_msteams)
> > > 
> > > 
> > > @gaby I know that, but in my use-case (running Uptime Kuma with Caprover) it would be more pratical if teams integration was built in.
> > > Where I work we have a total of 12 teams, my intention is to bring Kuma to be our main monitoring tool. We use Microsoft Teams to chat and each team have its own Microsoft Teams channel.
> > > My intention is to have all teams webhook url seted up and when adding a new monitor url, I just need to select the Microsoft Teams notification for the team responsible for that aplication.
> > > image
> > > Application	Team
> > > Application 01	Team A
> > > Application 02	Team A
> > > Application 03	Team B
> > > Application 04	Team C
> > > Application 05	Team C
> > > With Microsoft Teams integration built in I just need to create a webhook url for every new team and add it to Kuma notifications.
> > 
> > 
> > I see what you are trying to do, I just don't see how is that different than creating multiple notifications with apprise? Apprise is already built-in as you said, why should uptime-kuma create its own implementation of a notification already supported.
> 
> I just want Microsoft Teams, not other services, why not integrate it directly into uptime-kuma ?
> Teams is for professional use, and i think lot peoples are using it

If you are using Caprover, then you are using the Docker Image. Which means apprise/Microsoft Teams support is already built-in.

**Comment by gaby** [comment]
👍🏻 Looks good though!

File: src/components/notifications/Teams.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            >here</a>.
+        </div>
+    </div>
+</template>
+
+<script>
+export default {
+    data() {
+        return {
+            name: "teams",
```
Comment (Saibamen) [inline src/components/notifications/Teams.vue:25]: We can use `name` from `Teams` class

**Review by Saibamen — CHANGES_REQUESTED** [review]

**Comment by louislam** [comment]
Any existing notifications in Apprise will still be accepted, because Apprise is not available for non-docker users.

Also, native integration is usually faster while sending a notification and it is possible to have a better notification format. While Apprise could only provide a sentence of message.

**Comment by louislam** [comment]
Is it a paid feature? I cannot find the "App" tab in Microsoft Team

image

**Comment by WillianRod** [comment]
@louislam I think the free version does not allow teams creation. 

image

In the bottom I have a button called "Create or join a team" something like that.

I can help you test this implementation. I can create a Webhook URL so you can test.

Do you have somewhere I can send you that?

**Comment by WillianRod** [comment]
@louislam sent you you a webhook url on your email `info@website.louislam.net`

**Comment by louislam** [comment]
> @louislam sent you you a webhook url on your email `info@website.louislam.net`

Thanks, but not receive.
Never mind, I found an alumni ms account that enable to create a team in MS Teams.

Tested, UP/DOWN is ok, but general message is not sent.
That means if `heartbeatJSON` is null, it should send the `msg` .

**Comment by WillianRod** [comment]
@louislam cool, fixed 😄

**Pull request merged**